### DISCUSSION TITLE
Quesion about the appropriate tomograms for pytom picking

### DISCUSSION THREAD

**Discussion by PeterVanF** [SBC-Utrecht/pytom-match-pick#206] [Questions]
Hi,
I want to know whether we can use the NoisetoNoise denoised tomograms, like cryoCARE tomograms, for pytom template matching. There are different tomogram reconstruction methods such as WBP, SIRT, and SART. Which one is the best for pytom template matching? Additionally, what binning level should we use (in Å) for particle picking? Another question is whether there is a noticeable improvement in particle picking in 3D CTF-corrected tomograms, and under what circumstances should we start using 3D corrected tomograms for particle picking, considering that 3D CTF correction requires additional time to complete? If the particles are as large as ribosomes, is 3D CTF correction still necessary? Additionally, are the principles behind particle picking with pytom similar to the software stopgap, and is there a significant difference in performance between them?

**Comment by McHaillet** [comment]
Hi PeterVanF!

> I want to know whether we can use the NoisetoNoise denoised tomograms, like cryoCARE tomograms, for pytom template matching. There are different tomogram reconstruction methods such as WBP, SIRT, and SART. Which one is the best for pytom template matching?

For the classic reconstruction methods (WBP, SIRT, SART), I always recommend WBP. SIRT and SART put too much emphasis on low-frequency information which is not ideal for template matching. WBP has always produced the best results for me.

On denoising methods, it is hard to give a conclusive answer, as there are a lot of options. I tried using cryoCARE on WBP tomograms, but the results were not better than pure WBP (note to self: should put this in docs). It seems cryoCARE degrades high-resolution information too much, and is too aggressive: the cryoSAM rxiv paper illustrates this https://www.arxiv.org/pdf/2407.06833. I heard from other people that they tried IsoNet which also didnt work out. Perhaps newer methods like DeepDeWedge, or cryoSAM are better for this, but I haven't heard anyone try it.

In summary, my advice is too stick with WBP.

> Additionally, what binning level should we use (in Å) for particle picking?

Depends on particle size. For things like ribosomes/proteasomes ~15 Å is fine. Perhaps for smaller particles you could go to 10 Å, but I think it is difficult to exploit resolution beyond 1/30 Å<sup>-1</sup> (i.e. 15 Å pixels) in tomograms, unless you have isolated macromolecules. Usually the alignment of the tomogram is the limiting factor.

> Another question is whether there is a noticeable improvement in particle picking in 3D CTF-corrected tomograms, and under what circumstances should we start using 3D corrected tomograms for particle picking, considering that 3D CTF correction requires additional time to complete? If the particles are as large as ribosomes, is 3D CTF correction still necessary?

It's an optimization that gives slightly better results. For ribosomes you can easily get away without doing any sort of CTF correction to the tomogram. For 80S ribosome detection in my paper, I just put all the tomogram through AreTomo (version 1) and did the detection on those. It worked perfectly.

> Additionally, are the principles behind particle picking with pytom similar to the software stopgap, and is there a significant difference in performance between them?

I assume there is no big difference. But I haven't seen a direct comparison.

All the best!
Marten

**Comment by PeterVanF** [comment]
I got it. Thank you, Marten, for the detailed and very useful explanation!
All the best!
Peter

**Comment by alncat** [comment]
There is an automatic tomogram reconstruction software, Aretomo https://github.com/czimaginginstitute/AreTomo2. It can reconstruct tomogram using WBP in xyz order. I found it is very easy to use and can perform fiducialess alignment.

**Comment by shahpnmlab** [comment]
@McHaillet have you tried matching on tomograms generated from Warp/WArpTools? If I wanted to use those, would the "phase-flip" ctf model be accurate for these tomograms?

**Reply by McHaillet** [reply]
I haven't tried lately, but it should work fine. Indeed, you should use the 'phase-flip' option! This option just takes the absolute of the CTF and should work for tomograms that were ctf corrected via either phase flipping or ctf multiplication (which Warp does I believe). 

I think the CTF multiplication downweighs low resolution frequencies, so the matching were results will be more specific to high-res signal -- that could either be beneficial or not depending on the dataset!

**Reply by rdrighetto** [reply]
We are having good experience running `pytom-match-pick` with CTF-corrected tomograms generated by WarpTools (not denoised) 😉

**Comment by rdrighetto** [comment]
As for the reconstruction algorithm: please always use WBP for the reasons stated by @McHaillet above (beware that default in AreTomo is SART if I remember correctly)

Regarding using denoised tomograms: while you "can" use them, in my experience they don't work as well as using 3D CTF-corrected tomograms (generated by WarpTools, IMOD ctf3d or NovaCTF). Neural networks do weird stuff to the density values, and while they look good by eye (I always use them for visualization of the picks), they don't work so well for template matching. Picking ribosomes recently on DeepDeWedge-denoised tomograms I saw a lot of variation across the scores of true positives, while using non-denoised CTF-corrected tomos all ribos scored more or less equally well.
Using deconvolved tomograms as generated by WarpTools may work better than AI-denoised tomos because it's just a carefully crafted bandpass filter, but it's maybe too blurry already. Probably fine for ribosomes but not smaller stuff (have not tested).

As for the binning, we normally do everything in bin4 which corresponds to a pixel size of 7.68 Å/px but for challenging smaller targets bin2 tends to work better, as demonstrated in the works by Beata Turonova et al.
